### PR TITLE
bpo-43916: Rewrite new hashlib tests, fix typo (GH-25791)

### PR DESCRIPTION
**Pull request by tiran**
Signed-off-by: Christian Heimes <christian@python.org>

https://bugs.python.org/issue43916

### CODE REVIEW

File: Lib/test/test_hashlib.py [hunk 1]
```diff
@@ -901,36 +901,17 @@ def test_get_fips_mode(self):
...
-            constructors.append(_sha512.sha512)
-        except ImportError:
-            pass
-
-        for constructor in constructors:
-            h = constructor()
-            with self.subTest(constructor=constructor):
-                hash_type = type(h)
-                self.assertRaises(TypeError, hash_type)
+    def test_disallow_instantiation(self):
```
Comment (erlend-aasland) [inline Lib/test/test_hashlib.py:905]: Perhaps add `test.support.cpython_only` decorator?

Comment (tiran) [reply Lib/test/test_hashlib.py:905]: I assumed Victor didn't add the decorator for a reason.

Comment (erlend-aasland) [reply Lib/test/test_hashlib.py:905]: I'm guessing he forgot :) See Serhiy's [msg392169](https://bugs.python.org/issue43916#msg392169) in the bpo. Ditto for GH-25792.

**Review by erlend-aasland — APPROVED** [review]

File: Lib/test/test_hashlib.py (hunk 1 above)
Comment (tiran) [reply Lib/test/test_hashlib.py:905]: Thanks for the update!

**Pull request merged**